Extend RetryableException from Throwable interface

mitelg opened the pull request:
Let RetryableException extend from the Throwable interface to prevent errors in Psalm if catching RetryableException

<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no, I don't think so
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

We have some code which catches the `RetryableException`. Now Psalm complains about that catch because the `RetryableException` does not implement the `Throwable` interface

> Class/interface Doctrine\DBAL\Exception\RetryableException cannot be caught (see https://psalm.dev/132)

This is my first PR here. I hope I have done everything right. Looking forward for feedback :blush: :wave:

greg0ire commented:
Please do the same for `lib/Doctrine/DBAL/Driver/ExceptionConverterDriver.php`

greg0ire commented:
I think this can be considered as a bug. Can you also please annotate both interfaces with `@psalm-immutable`?

mitelg commented:
hey @greg0ire 
thanks for your feedback. did as you requested and also rebased the branch

mitelg commented:
it seems that ExceptionConverterDriver is not meant to be a throwable interface for exceptions.
I will revert that. Error in PHPStan:
> Child process error (exit code 255): PHP Fatal error:  Class Doctrine\DBAL\Driver\AbstractOracleDriver cannot implement interface Throwable, extend Exception or Error instead in  
     /home/mt/www/dbal/lib/Doctrine/DBAL/Driver/AbstractOracleDriver.php on line 15                                                                                                     
     Fatal error: Class Doctrine\DBAL\Driver\AbstractOracleDriver cannot implement interface Throwable, extend Exception or Error instead in                                            
     /home/mt/www/dbal/lib/Doctrine/DBAL/Driver/AbstractOracleDriver.php on line 15

mitelg committed:
Extend RetryableException from Throwable interface

Let RetryableException extend from the Throwable interface to prevent errors in Psalm if catching it

greg0ire commented:
Oh right, it has indeed nothing to do with exceptions, my bad. I'm closing and reopening to try to get the time sensitive test in Appveyor to pass.

SenseException commented:
Exception interfaces have the suffix `Exception` and since `DriverException` already implements `Throwable`, `RetryableException` seems to be the last one.

greg0ire commented:
Thanks @mitelg !

mitelg commented:
nice! thank you guys for accepting :blush:  :+1:
